File: About Codewars/Esolang Interpreters #1 – Introduction to Esolangs and My First Interpreter (MiniStringFuck)/main.erl
-module(mini_string_fuck).
-export([my_first_interpreter/1]).

my_first_interpreter(Code) ->
    interpreter(Code,0,[]).

interpreter(Code, Value, Output) ->
    case Code of
        [] -> Output;
        [Head | Tail] when Head == 43 -> interpreter(Tail, 
            case Value > 254 of
                true -> 0;
                _ -> Value + 1
            end ,Output);
        [Head | Tail] when Head == 46 -> interpreter(Tail,Value,lists:append(Output,[Value]));
        [_ | Tail] -> interpreter(Tail,Value,Output)
    end.

File: About Codewars/Esolang Interpreters #1 – Introduction to Esolangs and My First Interpreter (MiniStringFuck)/main.fs
let interpret (code: string) =
    let rec inter code value output =
        match code with
        | [] -> output
        | '+'::tail -> inter tail (match value > 254 with | true -> 0 | _ -> value + 1) output
        | '.'::tail -> inter tail value (((string)((char)value))::output)
        | _::tail -> inter tail value output
    (inter (code.ToCharArray() |> List.ofArray) 0 []) |> List.rev |> String.concat ""
